app/Http/Controllers/Auth/AdmissionController.php

```diff
@@ -165,54 +168,82 @@ public function updateNote(Request $request, Application $application): Redirect
         return redirect()->back();
     }
 
+    /**
+     * Show the finalize page with final names and statistics
+     */
+    public function indexFinalize(): View
+    {
+        $this->authorize('finalize', Application::class);
+        $admitted = Application::query()->with(['user', 'applicationWorkshops'])->admitted()->get()->sortBy('user.name');
+        $not_admitted = Application::query()->whereNotIn('id', $admitted->pluck('id'))->get();
+        return view('auth.admission.finalize', [
+            'semester' => $this->semester(),
+            'admitted_applications' => $admitted,
+            'users_to_delete' => User::query()
+                ->withoutGlobalScope('verified')
+                ->with('application')
+                ->whereIn('id', $not_admitted->pluck('user_id'))
+                //ignore users with any existing role
+                ->whereDoesntHave('roles')
```

Comment:
FYI: At the moment, even newly registered applicants get the printer role by default, so I don't think this will end up deleting anybody as-is. (last year's applicants are still present in the `users` table too)

https://github.com/EotvosCollegium/mars/blob/fabec6fb25c18d1c98a004e438094e62f5546fb2/app/Http/Controllers/Auth/RegisterController.php#L115

Probably easier to jump into the database and fix this manually, rather than changing the code here to handle this quirk.

Reply:
Huh, true... But this won't be the case after https://github.com/EotvosCollegium/mars/pull/394 has been merged. 
But even before that we can just delete that role.

Reply:
I agree; we don't need the printer role. No one can print without paying for admins before, anyway.